### PR TITLE
feat(issue-summary) Replace Google embeddings w/ local autofixability model

### PR DESCRIPTION
**Pull request by kddubey**
finetuned [bge-small](https://huggingface.co/BAAI/bge-small-en-v1.5) (132 MB) on our autofix eval issues + some synthetic negatives

the training code in **Local embeddings - ST collator** in [the training notebook](https://github.com/getsentry/data-analysis/blob/main/autofix/issue_summary/train_autofixability.ipynb) is close to adding a backwards pass to the zero-shot method we settled on for v1. does better than logistic regression on top of frozen Google embeddings

## Summary of results

- reduces p50 latency from 2.2 sec to 18 ms (on CPU)
- much steadier increase in precision as the score increases, according to held-out "realistic" issues and the `coding_was_run` proxy label. for example:
  - v1 hits 70% precision for the top 20% of issues, and 60% precision for the top 30%
  - v2 hits 70% precision for the top 40% of issues, and 60% precision for the top 45%.
- v2 may not be as robust to bigger changes in issue summary generation. it did well in the mix of OpenAI vs Gemini summaries in the held-out issues. for drastic issue summary prompting changes, this model would prolly need retraining using the notebook. can also consider retraining and re-evaluating when we have 👍 / 👎  data after a while.

### CODE REVIEW

File: src/seer/configuration.py [hunk 1]
```diff
@@ -112,6 +114,10 @@ def is_severity_enabled(self):
     def is_grouping_enabled(self):
         return self.GROUPING_ENABLED
 
+    @property
+    def is_autofixability_scoring_enabled(self):
+        return self.AUTOFIXABILITY_SCORING_ENABLED or "autofix" in self.HOSTNAME
```
[the remark this answers is not in the document]
Comment (jennmueng) [reply src/seer/configuration.py:119]: I'm not sure either, but should we be doing this rather than just asking SRE to set the env var?

[the remark this answers is not in the document]
Comment (kddubey) [reply src/seer/configuration.py]: [done](https://github.com/getsentry/ops/pull/14559)

**Review by jennmueng — APPROVED** [review]
